compiler/src/main/kotlin/motif/compiler/Dependencies.kt

```diff
@@ -60,15 +64,15 @@ class Dependencies private constructor(
             val nameScope = NameScope()
             val typeName = scopeImplTypeName.nestedClass("Dependencies")
 
-            val methods: SortedMap<Type, Method> = sinks
-                    .map { it.type }
-                    .toSet()
-                    .associateWith { type ->
-                        val methodSpec = methodSpec(nameScope, type)
+            val methods: SortedMap<Type, Method> = getRequestersByType(sinks)
```

Comment:
Instead of created these new classes, let's just do:

```kotlin
sinks
    .groupBy { it.type }
    .map { (type, sinks) ->
        val methodSpec = methodSpec(nameScope, type)
                .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
                .addJavadoc(getJavadoc(type, sinks))
                .build()
        Method(type, methodSpec)
    }
```
and...
```kotlin
private fun getJavadoc(type: Type, sinks: List<Sink>): CodeBlock {
    val codeBlock = CodeBlock.builder()
    // Javadoc logic in here
    return codeBlock.build()
}
```

Using a `CodeBlock` and making use of [JavaPoet's `$T` ](https://github.com/square/javapoet#t-for-types) should give us auto-imports for the javadoc types.

Reply:
That's much better! Completely forgot about $T's and JavaPoet's other features. That looks way better with all imports resolved.